fiberzap/zap_test.go

```diff
@@ -441,3 +441,26 @@ func Test_LoggerLevelsAndMessagesSingle(t *testing.T) {
 	utils.AssertEqual(t, levels[0], logs.All()[2].Level)
 	utils.AssertEqual(t, messages[0], logs.All()[2].Message)
 }
+
+// go test -run Test_Fields_Func
+func Test_Fields_Func(t *testing.T) {
+	app := fiber.New()
+	logger, logs := setupLogsCapture()
+
+	app.Use(New(Config{
+		Logger: logger,
+		FieldsFunc: func(c *fiber.Ctx) []zap.Field {
+			return []zap.Field{zap.String("test.custom.field", "test")}
+		},
+	}))
+
+	app.Get("/", func(c *fiber.Ctx) error {
+		return c.SendString("hello")
+	})
+
+	resp, err := app.Test(httptest.NewRequest("GET", "/", nil))
+
+	utils.AssertEqual(t, nil, err)
+	utils.AssertEqual(t, fiber.StatusOK, resp.StatusCode)
+	utils.AssertEqual(t, "test", logs.All()[0].Context[0].String)
```

Comment:
Can you add something like this to verify all the other expected fields are preset:

```go
expected := map[string]interface{}{
		"body":          "",
		"ip":            "0.0.0.0",
		"host":          "example.com",
		"url":           "/?foo=bar",
		"method":        "GET",
		"route":         "/",
		"protocol":      "http",
		"pid":           strconv.Itoa(os.Getpid()),
		"queryParams":   "foo=bar",
		"resBody":       "Cannot GET /",
		"bytesReceived": int64(0),
		"bytesSent":     int64(12),
	}
	utils.AssertEqual(t, expected, logs.All()[0].ContextMap())
```

Reply:
See latest commit, took inspiration from `Test_Logger_All`